pkg/search/postgres/store_test.go

```diff
@@ -452,6 +452,49 @@ func TestDeleteByQueryReturningIDs(t *testing.T) {
 	}
 }
 
+func TestPruneMany(t *testing.T) {
```

Comment:
I'm not sure how to actually trigger a "transient" batch failure. This is probably doable with a mock DB, but pretty cumbersome to implement.
I don't think the price to pay to have a more comprehensive test of the failure conditions is worth the gain on technical coverage. To me this test provides at least the needed smoke test coverage of the new function.

Reply:
The test behaviour looks very similar to the one of `TestDeleteMany`. Maybe the core of the test (using a key prefix parameter as well as a function to call DeleteMany/PruneMany on the store object) could be extracted and re-used to test DeleteMany/PruneMany in both generic and cached store variants.
This can be a follow-up papercut PR.